File: frontend/src/features/notifications/constants/NotificationPurposes.ts
/** Mirrors NotificationPurposes on the backend. Entry mode's own purpose (Analytic mode's
    field-purposes come from the analytic config instead): the fields whose values are listed
    out via the {fieldValueList} message token. */
export const NotificationPurposes = {
    Display: "Display",
} as const;

File: backend/src/Operum.Model/Constants/Notifications/NotificationPurposes.cs
namespace Operum.Model.Constants.Notifications
{
    // Entry mode's own purpose; Analytic mode uses AnalyticPurposes instead. Display fields are
    // listed via the {fieldValueList} message token.
    public static class NotificationPurposes
    {
        public const string Display = "Display";

        public static readonly HashSet<string> All = [Display];

        public static bool IsValid(string purpose) => All.Contains(purpose);
    }
}
